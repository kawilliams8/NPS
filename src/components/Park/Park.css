.Park {
  font-family: "Roboto Slab";
  background-image: url("../../assets/images/topo.png");
  border: .5px solid #989898;
  padding: 10px;
  margin: 28px;
}

.Park-card {
  border: 2px solid black;
  display: flex;
  margin: auto;
  width: 90%;
}

.Park-heading {
  font-size: 50px;
  margin: 10px;
  text-align: center;
  text-decoration: underline;
}

.Park-subheading {
  font-size: 21px;
  margin: 10px;
  text-align: center;
}

.Park-button-wrapper {
  display: flex;
  justify-content: space-evenly;
  margin: auto;
  width: 35%;
}

.Park-images-wrapper {
  display: flex;
  justify-content: space-evenly;
  margin: auto;
  width: 80%;
}

.Park-card-image {
  height: 400px;
  margin: 6px;
  object-fit: cover;
  width: 500px;
}

.Park-fave-button {
  color: #989898;
  font-family: "NationalPark-Regular";
  font-size: 18px;
  height: 36px;
  padding: 2px;
  margin: 6px;
  width: 200px;
}

.Park-fave-button:hover {
  color: black;
}

.Park-info-subheading {
  font-size: 18px;
}

.Park-info-cards {
  display: flex;
  justify-content: space-evenly;
}

.Park-card-info {
  background-color: #F5F5F5;
  border: .5px solid #989898;
  border-radius: 3px;
  font-size: 12px;
  margin: 10px 10px 10px 30px;
  padding: 10px;
  text-align: left;
  width: 45%;
}

.Park-site-link,
.Park-final-link,
.Park-weather-link {
  color: #585858;
  font-size: 16px;
}

.Park-site-link:hover,
.Park-final-link:hover,
.Park-weather-link:hover {
  color: #473328;
}

.Park-site-link:active,
.Park-final-link:active,
.Park-weather-link:active {
  color: #473328;
}

.Park-final-link {
  font-size: 28px;
  margin-left: 45%;
  width: 10%;
}