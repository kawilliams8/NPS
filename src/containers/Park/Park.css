.Park {
  border: black 2px solid;
  padding: 10px;
  margin: 20px;
  width: 400px;
}

.Park-card {
  border: 2px solid black;
  display: flex;
  margin: auto;
  width: 90%;
}

.Park-card-image {
  align-self: flex-start;
  height: 400px;
  margin: 10px;
  object-fit: cover;
  width: 500px;
}

.Park-card-map {
  align-self: flex-start;
  margin: 10px;
  width: 50%;
}

.Park-card-info {
  align-self: flex-end;
  margin: 10px;
  text-align: center;
  width: 50%;
}