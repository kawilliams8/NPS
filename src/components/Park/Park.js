import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { addFavorite, removeFavorite } from '../../actions';
import Trails from '../TrailsContainer/TrailsContainer';
import notFound from '../../assets/images/image-not-found.jpg';
import { key } from '../../containers/App/key';
import './Park.css';

export const Park = (props) => { 
  const { park } = props;
  let lat, long, mapUrl = "";
  if (park.latLong) {
    lat = park.latLong.split(" ")[0].substring(4);
    long = park.latLong.split(" ")[1].substring(5);
    mapUrl = `https://api.mapbox.com/styles/v1/mapbox/light-v9/static/${long},${lat}5,0,0/500x400?access_token=${key.mb_api_key}`
  }
  let img = notFound; 
  if (park.images) {
    img = park.images[0].url
  }

  return (
    <section className="Park">
      <h2 className="Park-heading">{park.name}</h2>
      <h3 className="Park-subheading">Full Name: {park.fullName}</h3>
      <h3 className="Park-subheading">Designation: {park.designation}</h3>
      <h3 className="Park-subheading">Home State: {park.states}</h3>
      <div className="Park-button-wrapper">
        <button className="Park-fave-button" onClick={() => props.addFavorite(park)}>Add to Favorites</button>
        <button className="Park-fave-button" onClick={() => props.removeFavorite(park)}>Remove from Favorites</button>
      </div>
      <div className="Park-images-wrapper">
        <img className="Park-card-image" src={img} alt={park.fullName} />
        {park.latLong && <img className="Park-card-image" src={mapUrl} alt='map' />}
      </div>
      <div className="Park-info-cards">
        <article className="Park-card-info">
          <h3 className="Park-info-subheading">Description:</h3>
          <h3>{park.description}</h3>
          <h3 className="Park-info-subheading">Directions:</h3>
          <h3>{park.directionsInfo}</h3>
          <h3 className="Park-info-subheading">Weather:</h3>
          <a className="Park-weather-link" href={park.weatherInfo} target="_blank" rel="noopener noreferrer">View Current Weather Report</a>
          <h3 className="Park-info-subheading">Further Information:</h3>
          <a className="Park-site-link" href={park.url} target="_blank" rel="noopener noreferrer">Official Site</a>
        </article>
        <Trails data={park}/>
      </div>
      <Link to="/parks" className="Park-final-link">RETURN</Link>
    </section>
  )
}

export const mapStateToProps = state => ({
  parks: state.parks,
  monts: state.monts,
  others: state.others,
  favorites: state.favorites
});

export const mapDispatchToProps = dispatch => ({
  addFavorite: park => dispatch(addFavorite(park)),
  removeFavorite: park => dispatch(removeFavorite(park))
});

export default connect(mapStateToProps, mapDispatchToProps)(Park);