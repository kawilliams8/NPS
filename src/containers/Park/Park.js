import React from 'react';
import { Link } from 'react-router-dom';
import notFound from '../../images/image-not-found.jpg';
import { key } from '../App/key';
import './Park.css';

const Park = ({ park }) => {
  console.log('inPark', park)
  let mapUrl = "";
  if (park.latLong) {
  const lat = park.latLong.split(" ")[0].substring(4);
  const long = park.latLong.split(" ")[1].substring(5);
  mapUrl = `https://api.mapbox.com/styles/v1/mapbox/light-v9/static/${long},${lat}5,0,0/500x400?access_token=${key.mb_api_key}`
  }

  return (
    <section className="Park-card">
      <img className="Park-card-image" src={park.images[0].url || notFound  } alt={park.fullName} />
      {park.latLong && <img className="Park-card-map" src={mapUrl} alt='map' />}
      <article className="Park-card-info">
      <h2>{park.name}</h2>
      <h3>{park.fullName}</h3>
      <h3>Designation: {park.designation}</h3>
      <h3>Home State: {park.states}</h3>
      <h3>{park.description}</h3>
      <h3>Directions: {park.directionsInfo}</h3>
      <h3>LatLong: {park.latLong}</h3>
      <h3>Weather: {park.weatherInfo}</h3>
      <a href={park.url} target="_blank" rel="noopener noreferrer">Official Site</a>
      <Link to="/parks" className="button">Go Back</Link>
      </article>
    </section>
  )
}

export default Park;