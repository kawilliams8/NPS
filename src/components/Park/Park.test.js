import React from 'react';
import { shallow } from 'enzyme';
import { Park } from '../Park/Park';

describe('Park', () => {

  let wrapper, mockPark, mockAddFavorite, mockRemoveFavorite;
  beforeEach(() => {
    mockPark = {
      "states": "DC",
      "directionsInfo": "The memorial is located at the corner of...",
      "directionsUrl": "http://www.nps.gov/afam/planyourvisit/directions.htm",
      "url": "https://www.nps.gov/afam/index.htm",
      "weatherInfo": "Washington DC gets to see all four seasons...",
      "name": "African American Civil War Memorial",
      "latLong": "lat:38.916554, long:-77.025977",
      "description": "Over 200,000 African-American soldiers...",
      "designation": "",
      "parkCode": "afam",
      "id": "1A47416F-DAA3-4137-9F30-14AF86B4E547",
      "fullName": "African American Civil War Memorial",
      "type": "parks",
      "images": [
        {
          "credit": "NPS Photo",
          "altText": "Site Statue",
          "title": "African American Civil War Statue",
          "id": "3406",
          "caption": "A poignant reminder of our nations past",
          "url": "https://www.nps.gov/common/uploads/structured_data/3C816B50-1DD8-B71B-0BF380049FB6B6A2.jpg"
        }
      ]
    }
    mockAddFavorite = jest.fn();
    mockRemoveFavorite = jest.fn();
    wrapper = shallow(<Park park={mockPark}
      addFavorite={mockAddFavorite}
      removeFavorite={mockRemoveFavorite}
      />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call the addFavorite method when clicked', () => {
    wrapper.find('button').at(0).simulate('click');
    expect(mockAddFavorite).toHaveBeenCalled();
  });

  it('should call the removeFavorite method when clicked', () => {
    wrapper.find('button').at(1).simulate('click');
    expect(mockRemoveFavorite).toHaveBeenCalled();
  });
})