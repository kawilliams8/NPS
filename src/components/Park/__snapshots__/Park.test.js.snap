// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Park should match the snapshot 1`] = `ShallowWrapper {}`;
